ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateArchive.java

```diff
@@ -94,71 +131,74 @@ public Optional<WorldState> get(final Hash rootHash, final Hash blockHash) {
 
   @Override
   public boolean isWorldStateAvailable(final Hash rootHash, final Hash blockHash) {
-    return trieLogManager.getBonsaiLayeredWorldState(blockHash).isPresent()
+    return trieLogManager.getBonsaiCachedWorldState(blockHash).isPresent()
         || persistedState.blockHash().equals(blockHash)
         || worldStateStorage.isWorldStateAvailable(rootHash, blockHash);
   }
 
-  public Optional<SnapshotMutableWorldState> getMutableSnapshot(final Hash blockHash) {
+  public Optional<MutableWorldState> getMutableSnapshot(final Hash blockHash) {
+    // TODO: decide whether we want to use BonsaiSnapshotWorldState or
+    // BonsaiSnapshotPersistedWorldState
     return rollMutableStateToBlockHash(
-            BonsaiSnapshotWorldState.create(this, worldStateStorage), blockHash)
+            //            BonsaiSnapshotWorldState.create(this, worldStateStorage), blockHash)
+            BonsaiSnapshotPersistedWorldState.create(this, worldStateStorage), blockHash)
```

Comment:
both implementations are working, but the 'persisted snapshot' version is currently enabled.  We need to do some performance and resource consumption analysis to see which one is "better" in practice